### PR TITLE
Refactor/icons package

### LINKED ISSUE #8160: Refactor crypto and token icons
The issue is to unify the API for icon components. The goal is to have all icons defined only in the icons package, create a universal component for rendering network and ERC20 token symbols, and a consistent wrapper for icons surrounded by a grey circle. These requirements would improve the consistency and maintainability of the codebase.

Acceptance criteria:
- Icon components are defined only in the `icons` package and nowhere else.
- There is a universal component that is able to render both network and ERC20 token symbols.
- In the design are many icons surrounded by a grey circle ([example](https://www.figma.com/file/Z6AGVUmKQzLNtDozFamW7f/%5Bs2.0%5D-Mobile?type=design&node-id=15635-69473&t=UtC5hhLTTXqmvuvO-4)).  A unified way of this wrapper is created, so it does not have to be redefined all over the codebase.

### PR DESCRIPTION
**Pull request by PeKne**
## Description
- icons package migrated from `@trezor/icons` to `@suite-common/icons`
- TokenIcon refactored so it's checking for the existence of the token icon itself internally
- new universal `CryptoIcon` which unifies API of `TokenIcon` and `CryptoIcon`
- new universal `RoundedIcon` which wraps any icon of token, crypto or normal icon into rounded container defined by design 

## Related Issue

Resolve #8160

### CODE REVIEW

File: suite-common/icons/src/components/CoinIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import { CryptoIconName } from '../icons';
+
+export type CoinSymbol = CryptoIconName | TokenSymbol;
+
+type CoinIconProps = {
+    symbol: CoinSymbol;
+    size?: CryptoIconSize;
+};
+
+export const CoinIcon = ({ symbol, size }: CoinIconProps) => {
```
Comment (Nodonisko) [inline suite-common/icons/src/components/CoinIcon.tsx]: Can we keep only rename this to CryptoIcon and basically merge tokens into it? I don't see reason why not and current state is really confusing. You may keep these components internally but they shouldn't be exposed out of package and probably should be grouped in folder or placed in CryptoIcon file directly.

**Review by Nodonisko — CHANGES_REQUESTED** [review]

File: suite-common/icons/src/components/CoinIcon.tsx (hunk 1 above)
Comment (PeKne) [reply suite-common/icons/src/components/CoinIcon.tsx]: Ok, I understood from the original PR where was `TokenIcon` introduced, that we want to have the network icons and token icons separated for easier further improvements. But I agree that this is way better from a developer experience perspective so I've reworked it to make only CryptoIcon exposed (2981f6c5f).

**Review by Nodonisko — APPROVED** [review]

**Review by juriczech — APPROVED** [review]

**Comment by bosomt** [comment]
**QA OK**


tested on latest dev build of Android app 429

**Pull request merged**